clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/MergeConflictResultWrapper.java

```diff
@@ -64,4 +67,31 @@ public ByteBuffer getUpdatedValueBytes() {
   public ByteBuffer getUpdatedRmdBytes() {
     return updatedRmdBytes;
   }
+
+  /**
+   * Return a best-effort value provider with the following behaviors:
+   *   1. returns the new value provider for PUT and UPDATE.
+   *   2. returns the old value for DELETE (null for non-existent key).
+   *   3. returns null if the value is not available.
```

Comment:
What is the case where the value would be totally not available?

Reply:
Today there's nothing to prevent users trying to delete a non-existent key. In that case the value would be totally not available.